docs/src/tutorial/pure.rst

```diff
@@ -209,6 +209,44 @@ Here is an example of a :keyword:`cdef` function::
         return a == b
 
 
+Managing the Global Interpreter Lock
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* ``cython.nogil`` can be used as a context manager or as a decorator to replace the :keyword:`nogil` keyword::
+
+    with cython.nogil:
+        # code block with the GIL released
+
+    @cython.nogil
+    @cython.cfunc
+    def func_released_gil() -> cython.int:
+        # function with the GIL released
+
+* ``cython.gil`` can be used as a context manager to replace the :keyword:`gil` keyword::
+
+    with cython.gil:
+        # code block with the GIL acquired
+
+  .. Note:: Cython currently does not support the ``@cython.gil`` decorator.
+
+When either ``cython.nogil`` or ``cython.gil`` is used as a function, it accepts a parameter for conditional
+releasing or acquiring the GIL. The condition must be constant (at compile time)::
```

Comment:
```suggestion
Both directives accept an optional boolean parameter for conditionally
releasing or acquiring the GIL. The condition must be constant (at compile time)::
```

Reply:
It would also be nice to have an actual usage example for something less trivial. Passing `True` and `False` makes it look useless.

Reply:
> EDIT: or mention at least fused types as a common use case, with a type based condition.

Added